sway-core/src/semantic_analysis/storage_only_types.rs

```diff
@@ -267,6 +267,21 @@ fn decl_validate(engines: Engines<'_>, decl: &ty::TyDeclaration) -> CompileResul
                             warnings,
                             errors
                         );
+                        for param in method.parameters {
+                            if param.name.as_str() != "self" {
```

Comment:
Maybe something like this?

```suggestion
                            if !param.is_self() {
```

Reply:
Thanks @anton-trunov , that works. I didn't know that the method existed.